src/Components/Web.JS/src/Platform/Mono/MonoTypes.d.ts

```diff
@@ -29,3 +29,19 @@ declare namespace MONO {
 // mono_bind_static_method allows arbitrary JS data types to be sent over the wire. However we are
 // artifically limiting it to a subset of types that we actually use.
 declare type BoundStaticMethod = (...args: (string | number | null)[]) => (string | number | null);
+
+declare namespace BINDING {
+  function js_string_to_mono_string(jsString: string): Pointer;
+  function js_typed_array_to_array(array: Uint8Array): Pointer;
+  function js_typed_array_to_array<T>(array: Array<T>): Pointer;
+  function conv_string(dotnetString: System_String | null): string | null;
+}
+
+// We don't actually instantiate any of these at runtime. For perf it's preferable to
+// use the original 'number' instances without any boxing. The definitions are just
+// for compile-time checking, since TypeScript doesn't support nominal types.
+declare interface MethodHandle { MethodHandle__DO_NOT_IMPLEMENT: any }
```

Comment:
I'm not entirely sure if this is still equivalent to https://github.com/dotnet/aspnetcore/pull/19544/files#diff-76be3ee9616d4fc0285c84959f51e891L27. Some of these types have to exist here for the typings in `Binding` to work correctly.

Reply:
Sorry, I don't understand the question. If there's some practical reason why these definitions need to be moved, let's discuss.